### PR TITLE
[cdc-common] add PublicEvolving annotation to SchemaUtils.

### PR DESCRIPTION
**Pull request by lvyanquan**
add PublicEvolving annotation to SchemaUtils, add modify code of SchemaManager to reuse the method of SchemaUtils.

@PatrickRen CC.

### CODE REVIEW

File: flink-cdc-common/src/main/java/com/ververica/cdc/common/utils/SchemaUtils.java [hunk 1]
```diff
@@ -32,6 +33,7 @@
 import java.util.stream.Collectors;
 
 /** Utils for {@link Schema} to perform the ability of evolution. */
+@PublicEvolving
```
Comment (PatrickRen) [inline flink-cdc-common/src/main/java/com/ververica/cdc/common/utils/SchemaUtils.java:36]: As we are promoting this utility to public, I think we need to be careful about the API, so I have concerns about the method `applySchemaChangeEvent` with parameter `ignoreException`. I prefer to remove this one, as I can't see any use cases for ignoring exceptions. Currently there's no usage of it, and the warning log is empty if the exception is ignored, which confuses me a bit. 

A better behavior would be clearly defining the exception on the method (`throws XXXException`) and letting the caller catch and handle it.

File: flink-cdc-runtime/src/main/java/com/ververica/cdc/runtime/operators/schema/coordinator/SchemaManager.java [hunk 2]
```diff
@@ -173,27 +167,8 @@ private void handleAlterColumnTypeEvent(AlterColumnTypeEvent event) {
...
-            }
-        }
-
-        // Dump the rest of information
-        schemaBuilder.primaryKey(oldSchema.primaryKeys());
-        schemaBuilder.comment(oldSchema.comment());
-        schemaBuilder.options(oldSchema.options());
-        Schema newSchema = schemaBuilder.build();
-        registerNewSchema(event.tableId(), newSchema);
+        registerNewSchema(
```
Comment (PatrickRen) [inline flink-cdc-runtime/src/main/java/com/ververica/cdc/runtime/operators/schema/coordinator/SchemaManager.java:170]: We can remove the if-else-if tree in `SchemaManager#applySchemaChange`, as `SchemaUtils#applySchemaChangeEvent` does the `instanceof` logic too.

**Review by PatrickRen — COMMENTED** [review]
@lvyanquan Thanks for the PR! I left some comments.

**Comment by lvyanquan** [comment]
Address it. And XXXException requires unified processing, so just keep to use IllegalArgumentException here.

**Review by PatrickRen — APPROVED** [review]
@lvyanquan Thanks for the update! LGTM.

**Pull request merged**